kahuna/public/js/search/results.js

```diff
@@ -400,21 +400,33 @@ results.controller('SearchResultsCtrl', [
         };
 
       const validatePhotoSalesSelection = (images) => {
-        let validImages = [];
-        let invalidImages = [];
-        images.forEach( (image) => {
-          if (image.data.usages.data.length === 0) {
-            validImages.push(image);
-          } else {
-            let syndicationExists = false;
-            for (const usage of image.data.usages.data) {
-              if (usage.data.platform === "syndication") {
-                syndicationExists = true;
-                break;
-              }
+        const validImages = [];
+        const invalidImages = [];
+
+        const filteredImages = images
+          .filter(image => {
+            if (image.data.uploadedBy === "Capture_AutoIngest") {
+              invalidImages.push(image);
+              return false;
+            }
+            return true;
+          })
+          .filter(image => {
+            if (image.data.usages.data.length === 0) {
+              validImages.push(image);
+              return false;
+            }
+            return true;
+          });
+        filteredImages.forEach((image) => {
+          let syndicationExists = false;
+          for (const usage of image.data.usages.data) {
+            if (usage.data.platform === "syndication") {
+              syndicationExists = true;
+              break;
```

Comment:
This function is really complex, and I think can be simplified a lot...

* I'd really caution against performing side effects like `.push()`ing while inside a filter function; it _does work_, but if not reading closely I'd assume that a filter doesn't change anything, and is only testing values against a predicate.
* The initial filter for `validImages` (testing if usages is 0) isn't really necessary; if you skipped that you should get the same result because the loop inside the later `forEach` wouldn't run, so `syndicationExists` would stay as `false` and the image would be added to `validImages`.
* The loop inside the `forEach` could be replaced by using a `.some`, which behaves the same but saves a couple of lines and avoids needing to track the `syndicationExists` variable.

Take a look at this I wrote, which I think would produce the same results, and see if it also looks clearer to you?

```js
const imageHasSyndicationUsage = (image) => {
  return image.data.usages.data.some(usage =>
    usage.data.platform === 'syndication'
  );
};

const validatePhotoSalesSelection = (images) => {
  const validImages = [];
  const invalidImages = [];

  images.forEach(image => {
    if (image.data.uploadedBy === 'Capture_AutoIngest' || imageHasSyndicationUsage(image)) {
      invalidImages.push(image);
    } else {
      validImages.push(image);
    }
  });

  return [validImages, invalidImages];
};

```

Reply:
Yeah that looks a lot better, cheers!